Contest_Control_System_Requirements.md

```diff
@@ -1222,66 +1225,6 @@ scoreboard-freeze-length: 1:00:00
 penalty-time:             20
 ```
 
-#### system.yaml
-
-A YAML file consisting of a mapping with the following keys:
-
-| Key             | Description                                                                                      |
-| --------------- | ------------------------------------------------------------------------------------------------ |
-| default-clars   | Sequence of pre-defined clarification answers. The first is the default and will be pre-selected |
-| clar-categories | Sequence of categories for clarifications.                                                       |
-| languages       | Sequence of mappings with keys as defined below                                                  |
-
-##### languages
```

Comment:
We're not actually dropping the language information, but this should instead be read from the CAF. That said, this section describes how languages are listed in system.yaml, so that should be removed, but maybe we should exchange it for a reference to the same information in the CAF at the same time?

Reply:
*should* or *could*? I mean, DOMjudge never read the language data from system.yaml and might not read the same data from the CAF, but we still work correctly.
We already link to the ICPC programming environment in [this section](https://ccs-specs.icpc.io/ccs_system_requirements#supported-languages), which list languages and compile options. We could add a sentence there that reads something like:

> The language options can also be read from the [Contest Archive Format](https://ccs-specs.icpc.io/contest_archive_format#language-object)

Reply:
> _should_ or _could_? 

Not sure... whatever it is now?

> I mean, DOMjudge never read the language data from system.yaml and might not read the same data from the CAF, but we still work correctly.

Yeah... I've always been of the opinion that humans can be part of the configuration system. I.e. if you look at the system.yaml file and then configure languages "manually", then you're "using" system.yaml. Same with the archive. It would be *very* nice if on top of that we could automatically *verify* that configuration has been done correctly, and including the compiler/interpreter information in the language endpoint is a big step in that direction.

> We could add a sentence there that reads something like:
> 
> > The language options can also be read from the [Contest Archive Format](https://ccs-specs.icpc.io/contest_archive_format#language-object)

I think we need to update the [Importing Contest Configuration section](http://localhost:4000/ccs_system_requirements#importing-contest-configuration). That does not *have* to be part of this PR though.